File: Java/src/Solution001_100/Solution94.java
package Solution001_100;

import tool.TreeNode;

import java.util.ArrayList;
import java.util.List;

/**
 * @author windows
 */
public class Solution94 {

    List<Integer> list = new ArrayList<>();
    public List<Integer> inorderTraversal(TreeNode root) {
        if (root == null){
            return list;
        }
        inorderTraversal(root.left);
        list.add(root.val);
        inorderTraversal(root.right);
        return list;
    }
}

File: C++/src/Solution1_100/Solution94.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution94.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

vector<int> Solution94::inorderTraversal(TreeNode *root) {
    vector<int> res;
    dfs(root, res);
    return res;
}

void Solution94::dfs(TreeNode *root, vector<int>& res){
    if (root == nullptr){
        return;
    }
    dfs(root->left, res);
    res.push_back(root->val);
    dfs(root->right, res);
}
